lib/experiments-page.php

```diff
@@ -271,23 +301,59 @@ function gutenberg_display_experiment_section() {
 	<?php
 }
 
-add_action( 'admin_init', 'gutenberg_handle_template_activate_setting_submission' );
-function gutenberg_handle_template_activate_setting_submission() {
+/**
+ * Handles form submission for the Gutenberg experiments settings page.
+ *
+ * This function processes the experiments form submission and handles:
+ * 1. Template activation setting (active_templates) - managed separately from regular experiments
+ * 2. Experiment dependency management - ensures sub-experiments are automatically disabled
+ *    when their parent experiments are disabled
+ */
+add_action( 'admin_init', 'gutenberg_handle_experiments_setting_submission' );
+function gutenberg_handle_experiments_setting_submission() {
```

Comment:
Piggy backing on this hook as a general experiments settings hook.

I'm using it to ensure that the inspector fields experiment is unset when the content only experiment is.

Reply:
> Does anything bad happen if the edit fields experiment is enabled and the content only experiment is disabled?

Not really I guess. The beef of the change checks for both: https://github.com/WordPress/gutenberg/pull/73605/files#diff-d193f64fcbd7c0c3b7c974f3768478220bf73faf5463233eb2274a1d0b3dbb11R18

We could check for both in the block library too?

Reply:
> We could check for both in the block library too?

I don't think we need both checks in the block library, though, as they're only used for the `window.__experimentalContentOnlyInspectorFields` experiment. If the latter is switched on and the former experiment isn't, does anything bad happen?

I guess what I'm trying to say is, the experiments _seem_ fairly isolated to me. Or do we never want to test the editable inspector fields separately from the logic that turns inserted patterns into contentOnly?

---

To try to use simpler words: is it useful to test editable content fields in isolation? My hunch was that it might be.